homeassistant/components/caldav/calendar.py

```diff
@@ -199,21 +199,47 @@ def update(self):
             dt.start_of_local_day(), dt.start_of_local_day() + timedelta(days=1)
         )
 
+        # Create new vevents for each recurrence of an event that happens today.
+        # For recurring events, some servers return the original event with recurrence rules
+        # and they would not be properly parsed using their original start/end dates.
+        vevents = [event.instance.vevent for event in results]
+        new_vevents = []
+        for vevent in vevents:
+            rrules = vevent.getrruleset()
+            if rrules:
+                occurrences = []
+                for rrule in rrules:
+                    if not rrule.tzinfo:
+                        rrule = dt.as_local(rrule)
+                    if (
+                        dt.start_of_local_day() + timedelta(days=1)
+                        > rrule
```

Comment:
When an event is all day, there is no timestamp portion. This shouldn't work right?

Reply:
There has to be a start time for an event, even if it's all day, no? That's all that `rrule` is.

Reply:
Nope. There is no timestamp part on all day so changing timezones doens't impact all day events. If your birthday is on Feb 14, it would not start Feb 13 at 11:00 if you were to move and change timezone

Reply:
OK, I may have fixed it in the latest version.